### PR TITLE
Core - renaming public APIs 

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/AEPSampleApp/extensions/EdgeIdentity.js [hunk 1]
```diff
@@ -13,30 +13,30 @@ governing permissions and limitations under the License.
 @format
 */
 
-import React, {Component} from 'react';
-import {Button, StyleSheet, Text, View, ScrollView} from 'react-native';
-import {AEPIdentity} from '@adobe/react-native-aepedgeidentity';
+import React, { Component } from 'react';
+import { Button, StyleSheet, Text, View, ScrollView } from 'react-native';
+import { Identity } from '@adobe/react-native-aepedgeidentity';
```
Comment (emdobrin) [inline apps/AEPSampleApp/extensions/EdgeIdentity.js:18]: this will be renamed in the edgeidentity branch, let's revert the changes for this file

File: packages/core/README.md [hunk 2]
```diff
@@ -190,11 +190,11 @@ var visitorId = new AEPVisitorID(idOrigin?: string, idType: string, id?: string,
...
 ```javascript
-AEPLifecycle.extensionVersion().then(version => console.log("AdobeExperienceSDK: AEPLifecycle version: " + version));
+Lifecycle.extensionVersion().then(version => console.log("AdobeExperienceSDK: Lifecycle version: " + version));
 ```
 
 ### [Signal](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/signals)
 ##### Getting the extension version:
 ```javascript
-AEPSignal.extensionVersion().then(version => console.log("AdobeExperienceSDK: AEPSignal version: " + version));
+Signal.extensionVersion().then(version => console.log("AdobeExperienceSDK: AEPSignal version: " + version));
```
Comment (emdobrin) [inline packages/core/README.md:199]: Signal version..

File: packages/core/js/index.d.ts [hunk 3]
```diff
@@ -33,47 +33,47 @@ export class AEPCore{
...
     static VERBOSE: string;
 }
-export class AEPMobileVisitorAuthenticationState{
+export class MobileVisitorAuthenticationState{
     static AUTHENTICATED: string;
     static LOGGED_OUT: string;
     static UNKNOWN: string;
 }
-export class AEPVisitorID{
+export class VisitorID{
```
Comment (emdobrin) [inline packages/core/js/index.d.ts:69]: Should we rename this to Identifiable/CustomIdentity as swift AEPCore? I feel like this can be confused with the VID. - this applies for the JS file too

File: packages/core/js/MobileCore.js [hunk 4]
```diff
@@ -136,22 +136,22 @@ module.exports = {
    * @param event Required parameter with {@link Event} instance to be dispatched. Should not be nil
    * @return true if the the event dispatching operation succeeded, otherwise the promise will return an error
    */
-  dispatchEvent(event: AEPExtensionEvent): Promise<boolean> {
+  dispatchEvent(event: Event): Promise<boolean> {
```
Comment (emdobrin) [inline packages/core/js/MobileCore.js:139]: This API can be renamed to `dispatch(event: Event)`

File: packages/core/js/MobileCore.js [hunk 5]
```diff
@@ -136,22 +136,22 @@ module.exports = {
...
    * <p>
    *
-   * @param event            required parameter, {@link AEPExtensionEvent} instance to be dispatched, used as a trigger
+   * @param event            required parameter, {@link Event} instance to be dispatched, used as a trigger
    * @param responseCallback required parameters, {@link Promise} to be called with the response event received
    *
    * @see AEPCore#dispatchResponseEvent(Event, Event, ExtensionErrorCallback)
    */
-  dispatchEventWithResponseCallback(event: AEPExtensionEvent): Promise<AEPExtensionEvent> {
+  dispatchEventWithResponseCallback(event: Event): Promise<Event> {
```
Comment (emdobrin) [inline packages/core/js/MobileCore.js:154]: can be renamed to: dispatchWithResponseCallback..

Unrelated to your changes, but I noticed this API overwrites the timeout to 1 sec in the objc implementation. Is this the desired behavior or can we fallback to the default in the native extension?

File: packages/core/js/MobileCore.js [hunk 6]
```diff
@@ -160,15 +160,15 @@ module.exports = {
...
    * @param requestEvent  required parameter, the event sent using
-   * 						{@link AEPCore#dispatchEventWithResponseCallback(AEPExtensionEvent)}
+   * 						{@link AEPCore#dispatchEventWithResponseCallback(Event)}
    * @return {@code boolean} indicating if the the event dispatching operation succeeded
    *
-   * @see AEPCore#dispatchEventWithResponseCallback(AEPExtensionEvent)
+   * @see AEPCore#dispatchEventWithResponseCallback(Event)
    */
-  dispatchResponseEvent(responseEvent: AEPExtensionEvent, requestEvent: AEPExtensionEvent): Promise<boolean> {
+  dispatchResponseEvent(responseEvent: Event, requestEvent: Event): Promise<boolean> {
```
Comment (emdobrin) [inline packages/core/js/MobileCore.js:171]: In Swift we don't have this API and you can register a listener + dispatch event, not sure about later versions of Android core, but  do we want to add this API in AEPCore?

File: packages/core/js/index.d.ts (hunk 3 above)
Comment (yangyansong-adbe) [reply packages/core/js/index.d.ts:69]:
```java
public static void getIdentifiers(final AdobeCallback<List<VisitorID>> callback);
```

```swift
static func getIdentifiers(completion: @escaping ([Identifiable]?, Error?) -> Void)
```

I think `Identifiable` is too _**Swifty**_, so I have to choose the class name we exposed in Android SDKs. IMO we'd better not pick a new class name for JS APIs.

File: packages/core/js/Identity.js [hunk 7]
```diff
@@ -58,13 +58,13 @@ module.exports = {
...
+   * If the current SDK privacy status is \ref PrivacyStatusOptOut, then this operation performs no action.
    *
    * @param identifiers a dictionary of customer IDs
-   * @param authenticationState a valid \ref AEPMobileVisitorAuthenticationState value.
-   * @see AEPMobilePrivacyStatus
+   * @param authenticationState a valid \ref MobileVisitorAuthenticationState value.
+   * @see PrivacyStatus
    */
-  syncIdentifiersWithAuthState(identifiers?: {string: string}, authenticationState: string) {
+  syncIdentifiersWithAuthState(identifiers?: { string: string }, authenticationState: string) {
```
Comment (emdobrin) [inline packages/core/js/Identity.js:67]: Any reason why `authenticationState` is not of type MobileVisitorAuthenticationState ? Same for the APIs below

File: packages/core/js/models/LogLevel.js [hunk 8]
```diff
@@ -20,7 +20,7 @@ const WARNING = "AEP_LOG_LEVEL_WARNING";
 const DEBUG = "AEP_LOG_LEVEL_DEBUG";
 const VERBOSE = "AEP_LOG_LEVEL_VERBOSE";
```
Comment (emdobrin) [inline packages/core/js/models/LogLevel.js:21]: can we rename all the constants too? e.g. AEP_LOG_LEVEL_VERBOSE -> VERBOSE
This comment applies for all enums

File: packages/core/js/MobileCore.js (hunk 6 above)
Comment (emdobrin) [reply packages/core/js/MobileCore.js:171]: Let's discuss about this and make any changes in a new PR.

File: packages/core/js/Identity.js (hunk 7 above)
Comment (yangyansong-adbe) [reply packages/core/js/Identity.js:67]: https://github.com/adobe/react-native-acpcore/blob/05a9b2ed10ccec9e7ae1693913884eaa03c6e060/js/ACPIdentity.js#L67
I just followed what we did in ACPIdentity wrappers. I'm not familiar with Identity extension, please feel free to caret a new PR to fix this issue.

File: packages/core/js/models/LogLevel.js (hunk 8 above)
Comment (yangyansong-adbe) [reply packages/core/js/models/LogLevel.js:21]: I think we'd better keep those AEP information, then it should be easier to identify wrapper logs in react native console.

File: packages/core/js/Identity.js (hunk 7 above)
Comment (emdobrin) [reply packages/core/js/Identity.js:67]: In my opinion we can use the enum here for the AEP wrapper. Would you revisit this in the final  review of the Core extensions?

File: packages/core/js/models/LogLevel.js (hunk 8 above)
Comment (emdobrin) [reply packages/core/js/models/LogLevel.js:21]: If the usage looks like `LogLevel.VERBOSE` or `PrivacyStatus.AUTHENTICATED` I feel the string constant will be more consistent with the usage that way if we don't have any other limitations.

File: packages/core/js/Identity.js (hunk 7 above)
Comment (yangyansong-adbe) [reply packages/core/js/Identity.js:67]: Emm ... I think we'd better let our Identity experts fix this issue in another PR.

File: packages/core/js/models/LogLevel.js (hunk 8 above)
Comment (yangyansong-adbe) [reply packages/core/js/models/LogLevel.js:21]: I agree an Enum should be a better fit here. I don't know if there is any limitation to it. Can you create a GitHub issue for it? I need to figure out why we use String instead of Enum here, then we can decide what we can do later.

File: packages/core/js/MobileCore.js (hunk 4 above)
Comment (yangyansong-adbe) [reply packages/core/js/MobileCore.js:139]: I think `dispatchEvent` is a better option for the loosely typed JavaScript. We also use this method name in Android SDK ->
```Java
public static boolean dispatchEvent(final Event event, final ExtensionErrorCallback<ExtensionError> errorCallback)
```

File: packages/core/js/MobileCore.js (hunk 5 above)
Comment (yangyansong-adbe) [reply packages/core/js/MobileCore.js:154]: https://github.com/adobe/aepsdk-core-ios/blob/daa7245798d99b9cc85f827a125d069aa7a54c29/AEPCore/Sources/core/MobileCore.swift#L118
`1 second` is the default timeout value of AEPCore API. We didn't overwrite it anyway.

File: packages/core/js/models/MobileVisitorAuthenticationState.js [hunk 9]
```diff
@@ -19,7 +19,7 @@ const AUTHENTICATED = "AEP_VISITOR_AUTH_STATE_AUTHENTICATED";
 const LOGGED_OUT = "AEP_VISITOR_AUTH_STATE_LOGGED_OUT";
 const UNKNOWN = "AEP_VISITOR_AUTH_STATE_UNKNOWN";
```
Comment (emdobrin) [inline packages/core/js/models/MobileVisitorAuthenticationState.js:20]: These ones need updates too

File: apps/AEPSampleApp/App.js [hunk 10]
```diff
@@ -27,7 +27,7 @@ function HomeScreen({ navigation }) {
       />
```
Comment (emdobrin) [inline apps/AEPSampleApp/App.js:27]: L17 requires update to `navigate('CoreView')`

File: apps/AEPSampleApp/extensions/IdentityView.js [hunk 11]
```diff
@@ -14,57 +14,57 @@ governing permissions and limitations under the License.
...
+  Identity.extensionVersion().then(version => console.log("AdobeExperienceSDK: Identity version: " + version));
 }
 function syncIdentifiers() {
-  AEPIdentity.syncIdentifiers({"id1": "identifier1"});
+  Identity.syncIdentifiers({ "id1": "identifier1" });
 }
 
 function syncIdentifiersWithAuthState() {
-  AEPIdentity.syncIdentifiersWithAuthState({"id1": "identifier1"}, "AEP_VISITOR_AUTH_STATE_AUTHENTICATED");
+  Identity.syncIdentifiersWithAuthState({ "id1": "identifier1" }, "AEP_VISITOR_AUTH_STATE_AUTHENTICATED");
```
Comment (emdobrin) [inline apps/AEPSampleApp/extensions/IdentityView.js:49]: we can use MobileVisitorAuthenticationState.AUTHENTICATED

File: packages/core/js/Identity.js (hunk 7 above)
Comment (emdobrin) [reply packages/core/js/Identity.js:67]: Sounds good, let's revisit this in a separate PR, I created https://github.com/adobe/aepsdk-react-native/issues/96

**Review by emdobrin — APPROVED** [review]
👍 Looks with with some small comments
